src/components/SwapWidget/SwapWidget.tsx

```diff
@@ -710,6 +654,98 @@ const SwapWidget: FC = () => {
     }
   };
 
+  useEffect(() => {
+    setAllowanceFetchFailed(false);
+    setBaseAmount(initialBaseAmount);
+    dispatch(clearTradeTerms());
+    unsubscribeFromGasPrice();
+    unsubscribeFromTokenPrice();
+    dispatch(clear());
+    LastLook.unsubscribeAllServers();
+  }, [
+    chainId,
+    dispatch,
+    LastLook,
+    unsubscribeFromGasPrice,
+    unsubscribeFromTokenPrice,
+  ]);
+
+  useEffect(() => {
+    if (ordersStatus === "reset") {
+      setIsApproving(false);
+      setIsSwapping(false);
+      setIsWrapping(false);
+      setIsRequestingQuotes(false);
+      setAllowanceFetchFailed(false);
+      setPairUnavailable(false);
+      setProtocolFeeInfo(false);
+      setShowGasFeeInfo(false);
+      setBaseAmount(initialBaseAmount);
+      LastLook.unsubscribeAllServers();
+    }
+  }, [ordersStatus, LastLook, dispatch]);
+
+  // Reset when the chainId changes.
+  useEffect(() => {
+    if (chainId) {
+      dispatch(resetOrders());
+    }
+  }, [chainId, dispatch]);
+
+  useEffect(() => {
+    setAllowanceFetchFailed(
+      allowances.swap.status === "failed" ||
+        allowances.wrapper.status === "failed"
+    );
+  }, [allowances.swap.status, allowances.wrapper.status]);
+
+  useEffect(() => {
+    if (!active) {
+      setShowTokenSelectModalFor(null);
+    }
+  }, [active]);
+
+  useEffect(() => {
+    if (!indexerUrls && library) {
+      dispatch(fetchIndexerUrls({ provider: library }));
+    }
+  }, [dispatch, indexerUrls, library]);
+
+  useEffect(() => {
+    if (indexerUrls) {
+      dispatch(
+        getFilteredOrders({
+          filter: {
+            senderTokens: [baseTokenInfo?.address!],
+            signerTokens: [quoteTokenInfo?.address!],
+            offset: 0,
+            limit: 100,
+          },
+        })
+      );
+    }
+  }, [baseTokenInfo, dispatch, indexerUrls, quoteTokenInfo]);
+
+  // check if serverURL query param exists
+  useEffect(() => {
+    if (queryUrl) {
+      setIsQueryingSelectedServer(true);
+      setServerUrl(queryUrl);
+    } else {
+      setIsQueryingSelectedServer(false);
+      setServerUrl(null);
```

Comment:
This will clear the serverUrl when order changes. Is this supposed to happen?

Reply:
I'll paste in the code because I made changes and the lines in the file are now different.

```if (queryUrl) {
      setIsQueryingSelectedServer(true);
      setServerUrl(queryUrl);
    } else {
      setIsQueryingSelectedServer(false);
      setServerUrl(null);
    }```

line 735 sets the value to false and line 736 to null because it's the `else` conditional to the previous block, which reads `if (queryURL) {`